src/LanguageServer/Impl/LanguageServer.cs

```diff
@@ -129,50 +129,12 @@ private void OnApplyWorkspaceEdit(object sender, ApplyWorkspaceEditEventArgs e)
             _idleTimeTracker?.NotifyUserActivity();
             using (await _prioritizer.DocumentChangePriorityAsync(cancellationToken)) {
                 var @params = ToObject<DidChangeTextDocumentParams>(token);
-                var version = @params.textDocument.version;
-                if (version == null || @params.contentChanges.IsNullOrEmpty()) {
-                    _server.DidChangeTextDocument(@params);
-                    return;
-                }
-
-                // _server.DidChangeTextDocument can handle change buckets with decreasing version and without overlapping 
-                // Split change into buckets that will be properly handled
-                var changes = SplitDidChangeTextDocumentParams(@params, version.Value);
-
-                foreach (var change in changes) {
-                    _server.DidChangeTextDocument(change);
-                }
-            }
-        }
-
-        private static IEnumerable<DidChangeTextDocumentParams> SplitDidChangeTextDocumentParams(DidChangeTextDocumentParams @params, int version) {
-            var changes = new Stack<DidChangeTextDocumentParams>();
-            var contentChanges = new Stack<TextDocumentContentChangedEvent>();
-            var previousRange = new Range();
-
-            for (var i = @params.contentChanges.Length - 1; i >= 0; i--) {
-                var contentChange = @params.contentChanges[i];
-                var range = contentChange.range.GetValueOrDefault();
-                if (previousRange.end > range.start) {
-                    changes.Push(CreateDidChangeTextDocumentParams(@params, version, contentChanges));
-                    contentChanges = new Stack<TextDocumentContentChangedEvent>();
-                    version--;
-                }
-
-                contentChanges.Push(contentChange);
-                previousRange = range;
+                _server.DidChangeTextDocument(@params);
             }
-
-            if (contentChanges.Count > 0) {
-                changes.Push(CreateDidChangeTextDocumentParams(@params, version, contentChanges));
-            }
-
-            return changes;
         }
 
         private static DidChangeTextDocumentParams CreateDidChangeTextDocumentParams(DidChangeTextDocumentParams @params, int version, Stack<TextDocumentContentChangedEvent> contentChanges)
             => new DidChangeTextDocumentParams {
-                _enqueueForAnalysis = @params._enqueueForAnalysis,
```

Comment:
Should this need to be removed from `DidChangeTextDocumentParams` while we're at it?

Reply:
Yeah, no references